base_name_search_improved/models/ir_model.py

```diff
@@ -20,35 +20,38 @@ def make_name_search():
             @api.model
             def name_search(self, name='', args=None,
                             operator='ilike', limit=100):
+                # Reusable parts
+                def extend_name_results(domain, results, limit):
+                    if len(results) < limit:
+                        domain += [('id', 'not in', [x[0] for x in results])]
+                        recs = self.search(domain, limit=limit - len(results))
+                        results.extend(recs.name_get())
+                    return results
                 # Regular name search
                 res = name_search.origin(
                     self, name=name, args=args, operator=operator, limit=limit)
-
+                # Extended name search is only used on some operators
                 allowed_ops = ['ilike', 'like', '=']
```

Comment:
Yes, and I could define the set outside this function, so that it's one less op to perform each time.